cypress/support/fragments/service_point/switchServicePoint.js

```diff
@@ -1,52 +1,40 @@
 import NewServicePoint from './newServicePoint';
-import { TextField, Dropdown, MultiColumnList, Button, Pane, Select, Modal, Accordion, HTML, including, MultiColumnListCell } from '../../../../interactors';
+import { TextField, Dropdown, MultiColumnList, Button, Pane, Select } from '../../../../interactors';
 import TopMenu from '../topMenu';
 import defaultUser from '../user/defaultUser';
 import permissions from '../../dictionary/permissions';
+import UsersSearchPane from '../users/usersSearchPane';
+import UsersEditPage from '../users/usersEditPage';
 
 export default {
   addServicePointPermissions: (username) => {
     cy.visit(TopMenu.usersPath);
-    cy.do([
-      TextField({ id: 'input-user-search' }).fillIn(username),
-      Button('Search').click(),
-      MultiColumnList().click({ row: 0, column: 'Active' }),
-      Pane({ id: 'pane-userdetails' }).find(Button('Actions')).click(),
-      Button({ id: 'clickable-edituser' }).click()]);
-    cy.intercept('/configurations/entries?query=(module==USERS%20and%20configName==custom_fields_label)').as('getPermissions1');
-    cy.intercept('/perms/permissions?length=10000&query=(visible==true)').as('getPermissions2');
-    cy.wait(['@getPermissions1', '@getPermissions2']);
-    cy.do([Button({ id: 'accordion-toggle-button-servicePoints' }).click(),
-      Button({ id: 'add-service-point-btn' }).click(),
-    ]);
-    cy.get('#list-column-selected').click();
-    cy.do([
-      Button({ id: 'save-service-point-btn' }).click(),
-      Select({ id: 'servicePointPreference' }).choose(NewServicePoint.defaultUiServicePoint.body.name),
-      Button({ id: 'accordion-toggle-button-permissions' }).click(),
-      Button({ id: 'clickable-add-permission' }).click(),
-      Modal({ id: 'permissions-modal' }).find(TextField({ type: 'search' })).fillIn(permissions.uiCheckinAll.gui),
-      Modal({ id: 'permissions-modal' }).find(Button('Search')).click(),
-      Modal({ id: 'permissions-modal' }).find(MultiColumnListCell(permissions.uiCheckinAll.gui)).click()]);
-    cy.expect(Modal({ id: 'permissions-modal' }).find(HTML(including('Total selected: 1'))));
-    cy.do(Button({ id: 'clickable-permissions-modal-save' }).click());
-    cy.expect(Modal({ id: 'permissions-modal' }).absent());
-    cy.expect(Accordion({ id: 'permissions' }).find(HTML(including(permissions.uiCheckinAll.gui))).exists());
-    cy.do(Button('Save & close').click());
-    cy.do(Button('User permissions').click());
+    UsersSearchPane.searchByKeywords(username);
+    UsersSearchPane.openUser(username);
+
+    UsersEditPage.addPermissions([permissions.checkinAll.gui]);
+    UsersEditPage.addServicePoints([NewServicePoint.defaultUiServicePoint.body.name]);
+    UsersEditPage.saveAndClose();
   },
 
-  logOutAndLogIn: ({ userName, password }) => {
+  logOutAndLogInByUser: ({ userName, password }) => {
```

Comment:
do you need this renaming? I suggest to have common logoutAndLogin() and more special version - logoutAndLoginAsAdmin()

Reply:
Done